Ginger/GingerCoreCommon/Repository/NewRepositorySerializer.cs

```diff
@@ -638,20 +612,69 @@ private static void xmlReadListOfObjects(object ParentObj, XmlReader xdr, IObser
                     {
                         return;
                     }
-
                 }
                 xdr.ReadEndElement();
             }
         }
 
+        private static void xmlReadListOfObjects_LazyLoad(MemberInfo mi, object parentObj, XmlReader xdr, IObservableList observableList)
+        {            
+            LazyLoadListConfig lazyLoadConfig = new LazyLoadListConfig();
+            lazyLoadConfig.ListName = mi.Name;
+            IsLazyLoadAttribute[] lazyAttr =null;
+            if (mi.MemberType == MemberTypes.Property)
+            {
+                lazyAttr = (IsLazyLoadAttribute[])((PropertyInfo)mi).GetCustomAttributes(typeof(IsLazyLoadAttribute), false);
+            }
+            else
+            {
+                lazyAttr = (IsLazyLoadAttribute[])((FieldInfo)mi).GetCustomAttributes(typeof(IsLazyLoadAttribute), false);
+            }
+            if (lazyAttr != null && lazyAttr.Length > 0)
+            {
+                lazyLoadConfig.LazyLoadType = lazyAttr[0].LazyLoadType;
+            }
+            else
+            {
+                lazyLoadConfig.LazyLoadType = LazyLoadListConfig.eLazyLoadType.StringData;//default
+            }
+            if (parentObj is RepositoryItemBase && lazyLoadConfig != null && observableList.AvoidLazyLoad == false)
+            {
+                observableList.LazyLoadDetails = new LazyLoadListDetails();
+                observableList.LazyLoadDetails.Config = lazyLoadConfig;
+                switch (observableList.LazyLoadDetails.Config.LazyLoadType)
+                {
+                    case LazyLoadListConfig.eLazyLoadType.NodePath:
+                        if (string.IsNullOrEmpty(((RepositoryItemBase)parentObj).FilePath) == false
+                                && File.Exists(((RepositoryItemBase)parentObj).FilePath)
+                                    && ((RepositoryItemBase)parentObj).DirtyStatus != Common.Enums.eDirtyStatus.Modified)
+                        {
+                            observableList.LazyLoadDetails.XmlFilePath = ((RepositoryItemBase)parentObj).FilePath;
+                            xdr.ReadOuterXml();//so xdr will progress
+                        }
+                        else //can't go with NodePath approch because no file to refernce or file do not have latest data
+                        {
+                            observableList.LazyLoadDetails.Config.LazyLoadType = LazyLoadListConfig.eLazyLoadType.StringData;
+                            observableList.LazyLoadDetails.DataAsString = xdr.ReadOuterXml();
+                        }
+                        break;
+
+                    case LazyLoadListConfig.eLazyLoadType.StringData:
```

Comment:
Codacy found an issue: [Remove this empty 'case' clause.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5848147)